### PR TITLE
refactor(gh-lines): use on demand ephemeral context menus

### PR DESCRIPTION
**Pull request by JPBM135**
Refactor the GH-lines to disable the auto-reply in favor of an ephemeral alternative.

- Added a cache layer to handle spikes in usage

Based (copied most parts) from Souji's branch https://github.com/almostSouji/discord-toolkit-bot/tree/github-ephem

### CODE REVIEW

File: src/commands/github.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+				files: this.resolveFiles(resolvedMatches, idx),
+			})),
+		);
+	}
+
+	private async reply(
+		interaction: InteractionParam<CommandMethod.MessageContext>,
+		payloads: InteractionReplyOptions[],
+	) {
+		this.cache.set(interaction.targetMessage.id, {
```
Comment (jaw0r3k) [inline src/commands/github.ts]: What if message gets edited?

**Comment by jaw0r3k** [comment]
And i still dont understand how is it better than just going to github page

File: src/commands/github.ts (hunk 1 above)
Comment (JPBM135) [reply src/commands/github.ts]: True 🤔

Comment (JPBM135) [reply src/commands/github.ts]: moved to an url based cache

**Comment by JPBM135** [comment]
> And i still dont understand how is it better than just going to github page

**Clicking:** Discord => github => wait github to load (Or open the browser) => goes back to discord
**Bot:** Discord => context menu

Not necessary but a nice optimization and its cool

**Comment by jaw0r3k** [comment]
> > And i still dont understand how is it better than just going to github page
> 
> **Clicking:** Discord => github => wait github to load (Or open the browser) => goes back to discord **Bot:** Discord => context menu
> 
> Not necessary but a nice optimization and its cool

Yup 
but most of ppl even dont know that it even exits

**Comment by JPBM135** [comment]
But we, cool people, do!!

Also with time it will spread

**Comment by JPBM135** [comment]
Idk, what you think @almostSouji ?

I like to have te option to do it in discord tbh

**Comment by almostSouji** [comment]
GH resolving was a fun idea for a thread use case but turned out to be an issue due to how threads are handled by discord. After some consideration, I really don't see much use in this being ephemeral, as the steps to get to the result are more elaborate than just clicking the link, which you might do anyways, eventually

**Comment by JPBM135** [comment]
It was good while it lasted, it will always be in our hearts 🫡

**Pull request closed without merging**